application/app/models/upload_collection.rb

```diff
@@ -70,7 +70,8 @@ def files
   end
 
   def connector_metadata
-    ConnectorClassDispatcher.connector_metadata(self)
+    #TODO refactor this
```

Comment:
What do you want to do here?

Reply:
This should point to a connector specific for upload_collection and not for download_collection